datacube_alchemist/worker.py

```diff
@@ -231,6 +213,101 @@ def post_messages(messages, count):
 
         return count
 
+    # Task related functions
+    def generate_task(self, dataset) -> AlchemistTask:
+        return AlchemistTask(dataset=dataset, settings=self.config)
+
+    def generate_task_by_uuid(self, uuid: str) -> AlchemistTask:
+        # Retrieve a task based on a UUID, or none if it doesn't exist for input product(s)
+        dataset = self._find_dataset(uuid)
+        if dataset:
+            return AlchemistTask(dataset=dataset, settings=self.config)
+        else:
+            return None
+
+    def generate_tasks(self, query, limit=None) -> Iterable[AlchemistTask]:
+        # Find which datasets needs to be processed
+        datasets = self._find_datasets(query, limit)
+
+        tasks = (self.generate_task(ds) for ds in datasets)
+
+        return tasks
+
+    # Queue related functions
+    def enqueue_datasets(self, queue, query, limit=None, product_limit=None):
+        datasets = self._find_datasets(query, limit, product_limit)
+        return self._datasets_to_queue(queue, datasets)
+
+    def find_fill_missing(self, queue, dryrun):
+        query = """
+            select d.id
+            from agdc.dataset as d
+            join agdc.dataset_type as t
+            on d.dataset_type_ref = t.id
+            left join (
+                select
+                    sub_s.dataset_ref,
+                    sub_s.source_dataset_ref
+                from agdc.dataset_source as sub_s
+                join agdc.dataset as sub_d
+                on sub_d.id = sub_s.dataset_ref
+                join agdc.dataset_type as sub_t
+                on sub_d.dataset_type_ref = sub_t.id
+                where sub_t.name = '{output_product}'
+            ) as s
+            on d.id = s.source_dataset_ref
+            where t.name in ({input_products})
+            and s.dataset_ref is null
```

Comment:
```suggestion
                where sub_t.name = %{output_product}s
            ) as s
            on d.id = s.source_dataset_ref
            where t.name in %{input_products}s
            and s.dataset_ref is null
```
Use SQL bindings instead of string formatting.